### ISSUE TITLE
error removing iam_instance_profile from launch configuration

### ISSUE THREAD

**Issue by singron** [hashicorp/terraform#3916]
I have a launch configuration like this

``` none
resource "aws_launch_configuration" "foo_lc" {
    image_id = "${var.ami}"
    spot_price = "0.35"
    instance_type = "r3.xlarge"
    security_groups = ["${var.security_group}"]
    iam_instance_profile = "iam_profile_to_remove"
    key_name = "${var.key_name}"
    user_data = "${template_file.foo_user_data.rendered}"
    lifecycle {
        create_before_destroy = true
    }
}
```

I accidentally added the iam_instance_profile and wanted to remove it from the launch configuration. At first, I just removed that whole line. But this gave the error:

``` none
Error applying plan:

1 error(s) occurred:

* aws_launch_configuration.foo_lc: doesn't support update
```

It's true that the launch configuration doesn't support update, but I have create_before_destroy set, so it should delete it and create a new one with the desired properties. To try and work around the issue, I instead set `iam_instance_profile = ""`. This created a plan which did delete the old launch configuration and create a new one, but when I tried to apply, I got a different error:

``` none
Error applying plan:

1 error(s) occurred:

* aws_launch_configuration.foo_lc: diffs didn't match during apply. This is a bug with Terraform and should be reported.
```

Then I tried the first solution again (just removing the iam_instance_profile), and it applied cleanly. This all happened on Terraform v0.6.6, although I set up the tfstate with Terraform v0.6.4 and updated to the v0.6.6 when I ran into the first issue. If I add the iam_instance_profile back in, I am able to replicate the same cycle of errors again.

**Comment by antonbabenko** [comment]
Same issue happens when trying to remove `spot_price` attribute from `aws_launch_configuration` - `aws_launch_configuration.foo_lc: doesn't support update`

**Comment by catsby** [comment]
Hey there - sorry the super long wait. This was finally fixed in https://github.com/hashicorp/terraform/pull/9699 and released in v0.7.8!
